### ISSUE TITLE
Docker module >=6 doesn't implement version_info

### ISSUE THREAD

**Issue by mkesper** [goldmann/docker-squash#211]
When using docker module 6.0.0 from pypi, docker-squash fails:

`2022-08-25 06:21:47,359 root         ERROR    module 'docker' has no attribute 'version_info'
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/docker_squash/image.py", line 357, in _save_image
    if docker.version_info[0] < 3:
AttributeError: module 'docker' has no attribute 'version_info'`

I opened an [issue](https://github.com/docker/docker-py/issues/3040) upstream but maybe docker-squash needs to guard against broken versions, too.

**Comment by NoamGoren** [comment]
the same issue for us, is there a possible workaround for the issue?

**Comment by T145** [comment]
Getting the same error from a latest `conda` environment running Python 3.10:

```
-> docker-squash -t ghcr.io/t145/black-mirror:squashed ghcr.io/t145/black-mirror
2022-08-26 19:52:34,656 root         INFO     docker-squash version 1.0.9, Docker a89b842, API 1.41...
2022-08-26 19:52:34,657 root         INFO     Using v2 image format
2022-08-26 19:52:34,671 root         INFO     Old image has 17 layers
2022-08-26 19:52:34,671 root         INFO     Checking if squashing is necessary...
2022-08-26 19:52:34,672 root         INFO     Attempting to squash last 17 layers...
2022-08-26 19:52:34,672 root         INFO     Saving image sha256:05669b7beaa29ab4fea5ca284712cb0e1f4e080803008d8ee3d162d947895ead to -\old directory...
2022-08-26 19:52:41,750 root         ERROR    module 'docker' has no attribute 'version_info'
Traceback (most recent call last):
  File "-\miniconda3\current\envs\squash\lib\site-packages\docker_squash\image.py", line 357, in _save_image
    if docker.version_info[0] < 3:
AttributeError: module 'docker' has no attribute 'version_info'
2022-08-26 19:52:41,751 root         WARNING  An error occured while saving the sha256:05669b7beaa29ab4fea5ca284712cb0e1f4e080803008d8ee3d162d947895ead image, retrying...
2022-08-26 19:52:41,752 root         INFO     Saving image sha256:05669b7beaa29ab4fea5ca284712cb0e1f4e080803008d8ee3d162d947895ead to -\old directory...
2022-08-26 19:52:48,807 root         ERROR    module 'docker' has no attribute 'version_info'
Traceback (most recent call last):
  File "-\miniconda3\current\envs\squash\lib\site-packages\docker_squash\image.py", line 357, in _save_image
    if docker.version_info[0] < 3:
AttributeError: module 'docker' has no attribute 'version_info'
2022-08-26 19:52:48,810 root         WARNING  An error occured while saving the sha256:05669b7beaa29ab4fea5ca284712cb0e1f4e080803008d8ee3d162d947895ead image, retrying...
2022-08-26 19:52:48,810 root         INFO     Saving image sha256:05669b7beaa29ab4fea5ca284712cb0e1f4e080803008d8ee3d162d947895ead to -\old directory...
2022-08-26 19:52:55,962 root         ERROR    module 'docker' has no attribute 'version_info'
Traceback (most recent call last):
  File "-\miniconda3\current\envs\squash\lib\site-packages\docker_squash\image.py", line 357, in _save_image
    if docker.version_info[0] < 3:
AttributeError: module 'docker' has no attribute 'version_info'
2022-08-26 19:52:55,963 root         WARNING  An error occured while saving the sha256:05669b7beaa29ab4fea5ca284712cb0e1f4e080803008d8ee3d162d947895ead image, retrying...
2022-08-26 19:52:55,965 root         ERROR    Couldn't save sha256:05669b7beaa29ab4fea5ca284712cb0e1f4e080803008d8ee3d162d947895ead image!
2022-08-26 19:52:55,966 root         ERROR    Execution failed, consult logs above. If you think this is our fault, please file an issue: https://github.com/goldmann/docker-squash/issues, thanks!
```

**Comment by rams3sh** [comment]
Facing similar issue in my environment as well. Log snippet below :- 

```
2022-08-28 15:40:28,881 root         INFO     docker-squash version 1.0.9, Docker 459d0df, API 1.41...
2022-08-28 15:40:28,881 root         INFO     Using v2 image format
2022-08-28 15:40:28,888 root         INFO     Old image has 26 layers
2022-08-28 15:40:28,888 root         INFO     Checking if squashing is necessary...
2022-08-28 15:40:28,888 root         INFO     Attempting to squash last 26 layers...
2022-08-28 15:40:28,888 root         INFO     Saving image sha256:859924c71662d8c80ae8c6c19a0477bed40651bf5cf9142b6dc9960d109fbcfa to /tmp/docker-squash-8g145e5j/old directory...
2022-08-28 15:40:31,238 root         ERROR    module 'docker' has no attribute 'version_info'
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/docker_squash/image.py", line 357, in _save_image
    if docker.version_info[0] < 3:
AttributeError: module 'docker' has no attribute 'version_info'
2022-08-28 15:40:31,239 root         WARNING  An error occured while saving the sha256:859924c71662d8c80ae8c6c19a0477bed40651bf5cf9142b6dc9960d109fbcfa image, retrying...
2022-08-28 15:40:31,239 root         INFO     Saving image sha256:859924c71662d8c80ae8c6c19a0477bed40651bf5cf9142b6dc9960d109fbcfa to /tmp/docker-squash-8g145e5j/old directory...
2022-08-28 15:40:33,573 root         ERROR    module 'docker' has no attribute 'version_info'
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/docker_squash/image.py", line 357, in _save_image
    if docker.version_info[0] < 3:
AttributeError: module 'docker' has no attribute 'version_info'
2022-08-28 15:40:33,573 root         WARNING  An error occured while saving the sha256:859924c71662d8c80ae8c6c19a0477bed40651bf5cf9142b6dc9960d109fbcfa image, retrying...
2022-08-28 15:40:33,573 root         INFO     Saving image sha256:859924c71662d8c80ae8c6c19a0477bed40651bf5cf9142b6dc9960d109fbcfa to /tmp/docker-squash-8g145e5j/old directory...
2022-08-28 15:40:35,937 root         ERROR    module 'docker' has no attribute 'version_info'
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/docker_squash/image.py", line 357, in _save_image
    if docker.version_info[0] < 3:
AttributeError: module 'docker' has no attribute 'version_info'
2022-08-28 15:40:35,937 root         WARNING  An error occured while saving the sha256:859924c71662d8c80ae8c6c19a0477bed40651bf5cf9142b6dc9960d109fbcfa image, retrying...
2022-08-28 15:40:35,938 root         ERROR    Couldn't save sha256:859924c71662d8c80ae8c6c19a0477bed40651bf5cf9142b6dc9960d109fbcfa image!
2022-08-28 15:40:35,938 root         ERROR    Execution failed, consult logs above. If you think this is our fault, please file an issue: https://github.com/goldmann/docker-squash/issues, thanks!
```

**Comment by shahedex** [comment]
There's a PR (#210) opened to resolve this issue.

For now, the following worked for me-
```
pip install docker==5
pip install docker-squash
```
The trick is not to use the latest version of `docker` package which will break `docker-squash`

**Comment by LeeXN** [comment]
docker-6.0.0 does not has version_info, You need uninstall it and install an old version
pip uninstall docker-6.0.0 \                                                                                     
pip install docker==5.0.3

**Comment by mkesper** [comment]
@shahedex @LeeXN I know you can work around the issue but still that's a workaround, not a longtime solution.

**Comment by goldmann** [comment]
Should be fixed with #216, thank you!
